services/sky/document_view.cc

```diff
@@ -209,7 +209,9 @@ mojo::Shell* DocumentView::GetShell() {
 
 void DocumentView::BeginFrame(base::TimeTicks frame_time) {
   if (sky_view_) {
-    sky_view_->BeginFrame(frame_time);
+    std::unique_ptr<LayerTree> layer_tree = sky_view_->BeginFrame(frame_time);
+    if (layer_tree->root_layer())
```

Comment:
Yeah, I wasn't sure what do in that case.  This happens if BeginFrame throws an exception and doesn't actually make a new LayerTree.  Previously, the UI thread would hold a ref to the SkPicture and just send it again, but the LayerTree is single-owner so the UI thread can't keep a reference.